pandas/core/index.py

```diff
@@ -1753,6 +1785,14 @@ def _get_nearest_indexer(self, target, limit):
         indexer = np.where(op(left_distances, right_distances)
                            | (right_indexer == -1),
                            left_indexer, right_indexer)
+        if tolerance is not None:
+            indexer = self._filter_indexer_tolerance(
+                target, indexer, tolerance)
+        return indexer
+
+    def _filter_indexer_tolerance(self, target, indexer, tolerance):
+        distance = abs(self.values[indexer] - target)
```

Comment:
I think you need to call ``_convert_tolerance`` here (you do this in ``tseries/base/get_loc`` ) so not sure which is the correct convention